### PR TITLE
Allow multiple edges with different type (catalog graph)

### PR DESCRIPTION
**Pull request by dweber019**
## Hey, I just made a Pull Request!

I'm currently implementing a new plugin, which is highly requested from end users in our firm.
Additionally, I saw some issues here, which go into the same direction like #17996 or #18751 

Here the first draft of my plugin code https://github.com/dweber019/backstage-plugins/tree/feat/relations/plugins/relations-backend

It works like intended and fits nicely our firms needs.

One thing I noticed, was that the catalog graph it not implemented with multiple edges in mind when the edges have the same source/from and target/to nodes but different relation types like (e.g. ownerBy, ledBy, applicationOwnerBy...).

What did I change.
1. The section in `if (unidirectional) {` will just add missing edges, if unidirectional is set to true, as with the implementation of this visitor pattern, we only visit nodes once and therefor are only picking up a relation once.
2. the `const finalEdges = edges.reduce((previousEdges` will merge relation types for edges so only edge remains per source and target. This change even seams to improve rendering performance when I tested this with a graph > 200 nodes.

Here some screenshots after the modification

Screen Shot 2024-03-11 at 22 50 53 PM
Screen Shot 2024-03-11 at 22 51 14 PM
Screen Shot 2024-03-11 at 22 51 26 PM
Screen Shot 2024-03-11 at 22 51 39 PM
Screen Shot 2024-03-11 at 22 52 11 PM

Would highly appreciate if we can bring this feature to the graph and grateful for any input on optimising my changes. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by dweber019** [comment]
@benjdlambert any chance to review there changes? would like to release the plugin, which needs this change to work properly 🙏 

https://github.com/dweber019/backstage-plugins/tree/main/plugins/relations-backend

**Comment by benjdlambert** [comment]
@dweber019 it needs a review from @backstage/catalog-maintainers before we can merge this. Pinging them internally too.

**Comment by dweber019** [comment]
@benjdlambert rebased on the latest and fixed DCO.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
nice thanks! sorry for the delay! :tada:

**Comment by dweber019** [comment]
Awesome thx @benjdlambert

**Pull request merged**
